enhancements/network/system-networkpolicies.md

```diff
@@ -0,0 +1,386 @@
+created the policy itself, it's possible the owner of the service
+might never realize (or might eventually forget) that this particular
+client was depending on it, and might change things in a way that
+would break that client.
+
+#### Pod-Level NetworkPolicies vs Namespace-Level NetworkPolicies
+
+Using the same example of a client Pod C in Namespace C owned by
+Operator C that needs to talk to a Service S in Namespace S owned by
+Operator S, what exactly should the NetworkPolicy say?
+
+The most-restrictive possibility would be to say "Pod C in Namespace C
+is allowed to connect to Service S in Namespace S". But it would be
+simpler and possibly more-future-proof to just say "All Pods in
+Namespace C can connect to all Pods in Namespace S".
+
+(In terms of efficiency, namespace-only rules are more efficient in
+openshift-sdn (since they result in a single match-on-VNID rule), but
+per-pod rules are generally more efficient in ovn-kubernetes (since
+namespace-based rules have to be rewritten in terms of individual pod
+IPs there).)
+
+Assuming the policy is going to be created by the operator managing
+the service, not by the operator managing the client, it may be best
+to be more restrictive on the service side, and less restrictive on
+the client side ("All Pods in Namespace C can connect to Service S in
+Namespace S"); that way if the client operator makes minor changes to
+its pods, it's less likely to require changes to a NetworkPolicy in
+another component.
+
+#### "Restrict when Restricted" vs "Allow when Allowed"
+
+Broadly speaking, there are two ways the policies could be written:
+
+  1. When the restricted mode is disabled, operators do nothing
+     special, just like now. When restricted mode is enabled,
+     operators create restrictive NetworkPolicies.
+
+  2. Operators always create restrictive NetworkPolicies, but when the
+     restricted mode is disabled, each system namespace also has a
+     permissive NetworkPolicy allowing all inbound access. (Because of
+     how NetworkPolicy works, the permissive policies would override
+     the restrictive ones).
+
+       a. The permissive NetworkPolicies could be created by the
+          individual operators along with the restrictive policies.
+
+       b. Alternatively, all of the permissive NetworkPolicies could
+          be created by the Cluster Network Operator.
+
+Option 2 is preferable to Option 1 because in general, NetworkPolicy
+is easier to work with when namespaces always have either a "default
+deny" or "allow to all" baseline policy. (See Appendix for gory
+details.)
+
+Of the two sub-options, 2b ("CNO creates permissive policies") seems
+nice because it means that individual operators wouldn't need to track
+the restricted-vs-open option; they'd just always configure themselves
+as restricted, and CNO would fix things up if they were supposed to be
+open.
+
+However, 2b would make upgrades more complicated, since we have to
```

Comment:
Do we _really_ need to worry about this? The only connections that would be disrupted would be 

1. Connections we somehow missed when writing network polices, and 
2. Only between operator S upgrade and CNO upgrade.

Reply:
I didn't mean "upgrades in general", I meant specifically the case of upgrading from a version of OCP that doesn't implement this feature, to the version of OCP that does, in a cluster where the administrator does not want to be using this feature, and has components of their own that we don't know about accessing random OpenShift components. In that case, during the upgrade, user workloads would be blocked from accessing OpenShift components, possibly creating outages.

If you're going to argue that that's not a real problem, then you're essentially arguing that we don't actually need to preserve the permissive option at all; we can just start switching all components to be restrictive with no way to override it.